src/format/mod.rs

```diff
@@ -557,15 +578,32 @@ fn format_inner<'a>(
                 result: &mut String,
                 off: FixedOffset,
                 allow_zulu: bool,
-                use_colon: bool,
+                colon_type: Option<ColonType>,
```

Comment:
I think this should not take an `Option`, just make it `Colons` directly (requiring the callers to pass `Colons::None`).